src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,505 @@
+        trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            trade_history.update(self._process_trade_history(index))
+            index += _TRADE_RECORD_LIMIT
+
+        # reset index and count for next API call
+        index = 0
+        count = int(self._client.private_post_ledgers(params={_OFFSET: index})[_RESULT][_COUNT])
+        ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            ledger.update(self._process_ledger(index))
+            index += _TRADE_RECORD_LIMIT
+
+        result = (trade_history, ledger)
+
+        if self.use_cache:
+            save_to_cache(self.__CACHE_FILE, result)
+
+        return result
+
+    def load(self) -> List[AbstractTransaction]:
+        (trade_history, ledger) = self._gather_api_data()
+
+        result: Dict[str, List[AbstractTransaction]] = {'in': [], 'out': [], 'intra': []}
+
+        unhandled_types: Dict[str, str] = {}
+        for key, value in ledger.items():
+            if value[_TYPE] == _WITHDRAWAL or value[_TYPE] == _DEPOSIT:
+                result[_INTRA].append(key)
+            elif value[_TYPE] == _TRADE and 'USD' not in value[_ASSET]:
+                if RP2Decimal(value[_AMOUNT]) > ZERO: result[_IN].append(key)
+                else: result[_OUT].append(key)
+            elif value[_TYPE] == _MARGIN:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _ROLLOVER:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _TRANSFER:
+                result[_IN].append(key)
+            elif value[_TYPE] == _SETTLED:
+                # ignorable in terms of in/out/intra
+                pass
+            elif value[_TYPE] == _CREDIT:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _STAKING:
+                # 'staking not implemented'
+                pass
+            elif value[_TYPE] == _SALE:
+                # 'sale not implemented'
+                pass
+            else:
+                unhandled_types.update({value[_TYPE]: key})
+        self.__logger.debug(f"unhandled types of the ledger={unhandled_types}")
+
+        return self._compute_tx_set(trade_history, ledger, result)
+
+    def _compute_tx_set(self, trade_history, ledger, processed_transactions) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        transactions: List[AbstractTransaction] = processed_transactions[_IN] + \
+                                                  processed_transactions[_OUT] + \
+                                                  processed_transactions[_INTRA]
+        for key in transactions:
+            record: Dict[str, str] = ledger[key]
+            self.__logger.debug("Ledger record: %s", record)
+
+            timestamp_value: str = self._rp2_timestamp_from_ms_epoch(str(RP2Decimal(record[_TIMESTAMP])*RP2Decimal(_MS_IN_SECOND)))
```

Comment:
Once you make the change I mentioned above with `abstract_ccxt_input_plugin.py` you can shorten this down and make it more manageable:
```
timestamp_value: str = self._rp2_timestamp_from_epoch(record[_TIMESTAMP])
```

No point in doing the extra math.